ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp

```diff
@@ -324,35 +330,50 @@ class FoxgloveBridge : public rclcpp::Node {
     // Create a QoS profile that only keeps the last message since we don't need to keep a history,
     // and use reliable delivery if all publishers are reliable otherwise best-effort
     bool reliable = true;
+    bool durationVolatile = true;
+    size_t depth = 1;
     for (const auto& publisher : this->get_publishers_info_by_topic(topic)) {
-      if (publisher.qos_profile().reliability() == rclcpp::ReliabilityPolicy::BestEffort) {
+      const auto& qos = publisher.qos_profile();
+      if (qos.reliability() == rclcpp::ReliabilityPolicy::BestEffort) {
         reliable = false;
-        break;
       }
+      if (qos.durability() == rclcpp::DurabilityPolicy::TransientLocal) {
+        durationVolatile = false;
+      }
+      depth += qos.depth();
```

Comment:
```suggestion
      constexpr size_t MAX_QUEUE_DEPTH = 10;
      depth = std::min(MAX_QUEUE_DEPTH, std::max(depth, qos.depth()));
```

Reply:
I tested it locally with having 4 transient local publishers and it only works if the depth is at least as big as the amount of publishers.

```bash
# Launch 4 publishers
for i in {1..4}; do ros2 topic pub --qos-history keep_last --qos-depth 1 --qos-durability transient_local -1 --keep-alive 100 /foo std_msgs/String 'data: "hi"' & done
```

```bash
# Subscribe with depth = 1
$ ros2 topic echo --qos-durability transient_local --qos-depth 1 --qos-reliability reliable /foo
data: hi
---
```



```bash
# subscribe with default depth (not sure what the default is)
$ ros2 topic echo --qos-durability transient_local --qos-reliability reliable /foo
data: hi
---
data: hi
---
data: hi
---
data: hi
---
```

Reply:
Maybe we can make it the following?

```cpp
constexpr size_t MAX_QUEUE_DEPTH = 10;
depth = std::min(MAX_QUEUE_DEPTH, depth + qos.depth());

Reply:
Maybe the best we could do is to replicate how `ros2 topic echo` chooses the subscriber qos?

https://github.com/ros2/ros2cli/blob/619b3d1c9b38f49bfd2a9286b1086d8eed1ecb94/ros2topic/ros2topic/verb/echo.py#L137-L194

Reply:
`ros2 topic echo` uses the [qos_profile_sensor_data](https://github.com/ros2/rmw/blob/31c6fd094c8bd01d0a231856df1bd9a476bea26a/rmw/include/rmw/qos_profiles.h#L25-L36) qos preset which has a default depth of 5

Reply:
It probably makes sense to add a parameter that allows to choose one of the [qos presets](https://github.com/ros2/rmw/blob/31c6fd094c8bd01d0a231856df1bd9a476bea26a/rmw/include/rmw/qos_profiles.h) as the default subscription qos profile